### PR TITLE
Add pacman

### LINKED ISSUE #8734: Create pacman.rb
**Describe the bug**
Error when trying to install pacman

**To Reproduce**

1. I executed this command:

      ```shell
      cd /usr/local/lib/crew/packages
      sudo nano [pacman.rb](https://github.com/chromebrew/chromebrew/files/12801080/pacman.txt)
      ```

      Output:

      ```
      [pacmanOutput.txt](https://github.com/chromebrew/chromebrew/files/12801098/pacmanOutput.txt)
      ```

**Expected behavior**
expected it to install pacman

**Environment: Paste the output of `crew sysinfo -v` between the two `***` below**

***
<details><summary>Expand</summary>

- Architecture: `x86_64` (`x86_64`)
- Processor vendor: `GenuineIntel`
- User space: `64-bit`
- Chromebrew Kernel version: `4.19`
- Chromebrew Running in Container: `false`

- Chromebrew version: `1.35.6`
- Chromebrew prefix: `/usr/local`
- Chromebrew libdir: `/usr/local/lib64`

- Last update in local repository: 72fe3f94 `Xorg Security Updates: CVE-2023-43785 (#8732) (3 hours ago)`

- OS variant: `Chrome OS`
- OS version: `cyan-release/R103-14816.131.0`
- OS channel: `stable-channel`

</details>

also the original package was for `https://github.com/willtrnr/capman/blob/master/pacman.rb` and i edited it to a newer version with compatibilty field
Please help

### PR DESCRIPTION
**Pull request by satmandu**
Fixes #8734
- Based upon work by @Kalion2000

Builds properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_REPO=https://github.com/satmandu/chromebrew.git CREW_BRANCH=pacman crew update
```

### CODE REVIEW

File: packages/pacman.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    system 'curl -Lf https://archlinux.org/mirrorlist/all/ -o mirrorlist'
+    # Uncomment worldwide mirror in mirror list.
+    system "sed -i ':a;N;\$!ba;s/## Worldwide\\\n#Server/## Worldwide\\\nServer/' mirrorlist"
+    FileUtils.install 'mirrorlist', "#{CREW_PREFIX}/etc/pacman.d/mirrorlist", mode: 0o644
+    open("#{CREW_META_PATH}/pacman.directorylist", 'a') do |f|
+      f.puts "#{CREW_PREFIX}/etc/pacman.d"
+    end
+    open("#{CREW_META_PATH}/pacman.filelist", 'a') do |f|
+      f.puts "#{CREW_PREFIX}/etc/pacman.d/mirrorlist"
+    end
```
Comment (uberhacker) [inline packages/pacman.rb:70]: Shouldn't this be in `def self.build` so it can be uninstalled?

File: packages/pacman.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      f.puts "Include = #{CREW_PREFIX}/etc/pacman.d/mirrorlist"
+      f.puts '[extra]'
+      f.puts "Include = #{CREW_PREFIX}/etc/pacman.d/mirrorlist"
+      f.puts '[community]'
+      f.puts "Include = #{CREW_PREFIX}/etc/pacman.d/mirrorlist"
+    end
+    # Figure out how to enable trusts for the gpg sigs...
+    system "sed -i '/#SigLevel/a SigLevel = TrustAll' #{CREW_PREFIX}/etc/pacman.conf"
+    puts 'Installing pacman mirrorlist.'.lightblue
+    system 'curl -Lf https://archlinux.org/mirrorlist/all/ -o mirrorlist'
```
Comment (uberhacker) [inline packages/pacman.rb:61]: Use downloader with sha256 checksum?

Comment (satmandu) [reply packages/pacman.rb:61]: The mirrorlist will change, but I have changed this to use downloader.

**Comment by satmandu** [comment]
Closing as this is a Proof of Concept, but this WILL conflict with Chromebrew installs if one installs any packages with this.

**Comment by yeesterPlus** [comment]
M A N

**Pull request closed without merging**